### PR TITLE
Support throwing exception in Java TestProbe.awaitAssert

### PR DESCRIPTION
**Pull request by adrian-2414745**
#29677

How to fix bin compat ?
[error] (akka-actor-testkit-typed / mimaReportBinaryIssues) Failed binary compatibility check against com.typesafe.akka:akka-actor-testkit-typed_2.12:2.6.8! Found 9 potential problems

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Review by johanandren — COMMENTED** [review]
We don't keep binary compatibility for the testkit, and this should be source compatible (well, as long as you use lambda syntax rather than pass an instance) so I think this is fine to add mima filters for, you can read how to add a filter here: https://github.com/akka/akka/blob/master/CONTRIBUTING.md#binary-compatibility

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala [hunk 1]
```diff
@@ -339,13 +340,13 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
   override def awaitAssert[A](a: => A): A =
     awaitAssert_internal(a, remainingOrDefault, 100.millis)
 
-  override def awaitAssert[A](max: JDuration, interval: JDuration, supplier: Supplier[A]): A =
-    awaitAssert_internal(supplier.get(), max.asScala.dilated, interval.asScala)
+  override def awaitAssert[A](max: JDuration, interval: JDuration, supplier: Creator[A]): A =
```
Comment (octonato) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala:343]: rename arg `supplier` to `creator` here and in the subsequent cases.

**Review by octonato — APPROVED** [review]
Looks good to me, thanks @adrian-salajan. 

with a minor change request, see below.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**